rxandroidble/src/main/java/com/polidea/rxandroidble/internal/operations/RxBleRadioOperationScan.java

```diff
@@ -46,8 +46,11 @@ public void run() {
     }
 
     public void stop() {
+
         // TODO: [PU] 29.01.2016 https://code.google.com/p/android/issues/detail?id=160503
-        rxBleAdapterWrapper.stopLeScan(leScanCallback);
+        if (startLeScanStatus) {
```

Comment:
Still there is room for a race condition. When the radio will start executing the scan operation but will not yet call `run()` and the operation would be unsubscribed then.
We could check if the operation wasn't unsubscribed right before calling `startLeScan()` but it would still need some manual synchronisation.